spec/blackhole_spec.rb

```diff
@@ -11,4 +11,8 @@
     expect(null.foobaz).to be(null)
     expect(null << 'bar').to be(null)
   end
+
+  it 'can be a hash value when serialized as yaml' do
+    expect({null: null}.to_yaml).to eq("---\n:null: !ruby/object {}\n")
+  end
```

Comment:
I'm not sure what the ideal return value for `#to_int` is, but `nil` doesn't work and `0` doesn't seem to have any ill-effects.

@tenderlove any thoughts on the meaning of `define_method(:to_int) { 0 }` on serialization of `{null: null}.to_yaml`?

Ref:  [Psych::Visitors::YAMLTree](https://github.com/ruby/ruby/blob/v2_2_0/ext/psych/lib/psych/visitors/yaml_tree.rb)

```ruby
        def key? target
          @obj_to_node.key? target.object_id
        rescue NoMethodError
          false
        end
```

should psych also rescue a TypeError, instead of just NoMethodError?